deno2/main.rs

```diff
@@ -0,0 +1,39 @@
+extern crate libc;
+use std::ffi::CStr;
+use std::ffi::CString;
+use libc::c_char;
+use libc::c_int;
+
+#[link(name = "deno", kind = "static")]
+extern "C" {
+    fn deno_v8_version() -> *const c_char;
+    fn deno_init();
+    fn deno_set_flags(argc: *mut c_int, argv: *const *const c_char);
+}
+
+fn set_flags() {
+    // Create a vector of zero terminated strings.
+    let args = std::env::args()
+        .map(|arg| CString::new(arg).unwrap())
+        .collect::<Vec<CString>>();
+    // Convert the strings to raw pointers
+    let argv = args.iter()
+        .map(|arg| arg.as_ptr())
+        .collect::<Vec<*const c_char>>();
+    let mut argc = argv.len() as c_int;
+    let argc_ptr = &mut argc as *mut c_int;
+    unsafe {
+        // pass the pointer of the vector's internal buffer to a C function
+        deno_set_flags(argc_ptr, argv.as_ptr());
```

Comment:
I'm pretty sure you can pass `&mut argc` here, there's need to convert it to a raw pointer explicitly.

Reply:
That doesn't work.

Reply:
Could you elaborate on how exactly it didn't work? I've just tried that and it compiles just fine.

```diff
     let mut argc = argv.len() as c_int;
-    let argc_ptr = &mut argc as *mut c_int;
     unsafe {
         // pass the pointer of the vector's internal buffer to a C function
-        deno_set_flags(argc_ptr, argv.as_ptr());
+        deno_set_flags(&mut argc, argv.as_ptr());
     };
 }
```

Reply:
Ah, sorry, I changed it to &mut argv. Oops.

Reply:
It does work. Thanks for the suggestion @aqrln.